From: Rob Browning <rlb@defaultvalue.org>
Date: Sun, 5 Feb 2023 07:04:15 +0900
Subject: add-unix-to-cpp-undefs

* ./configure should no longer define "unix" when building Emacs.
  Patch: add-unix-to-cpp-undefs.diff
  Date: Sun May  2 07:23:27 UTC 2010
  Added-by: Rob Browning <rlb@defaultvalue.org>
  Status: incorporated upstream

  Before this fix, gcc would produce the following warning:

    gcc: unrecognized option '-2.0/'

  The Debian patch is taken from this upstream commit:

    revno: 99984
    committer: Miles Bader <miles@gnu.org>
    branch nick: trunk
    timestamp: Thu 2010-04-22 09:10:30 +0900
    message:
      get rid of "unix" pre-defined macro when preprocessing Makefile
---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 09b1e76..aef0767 100644
--- a/configure.in
+++ b/configure.in
@@ -3343,7 +3343,7 @@ test "${exec_prefix}" != NONE &&
 # the C preprocessor to some helpful value like 1, or maybe the empty
 # string.  Needless to say consequent macro substitutions are less
 # than conducive to the makefile finding the correct directory.
-[cpp_undefs="`echo $srcdir $configuration $canonical |
+[cpp_undefs="`echo $srcdir $configuration $canonical unix |
   sed -e 's/[^a-zA-Z0-9_]/ /g' -e 's/^/ /' -e 's/  *$//' \
   -e 's/  */ -U/g' -e 's/-U[0-9][^ ]*//g'`"]
 
